Fix flaky unseal_and_spill_test  test

### Describe the bug

```

    inlined from ‘std::pair<{anonymous}::item_initialization, std::__cxx11::basic_string<char> > {anonymous}::scenario_initial_state(scenario)’ at ../tests/unseal_and_spill_test.cpp:365:29:
Error: ../tests/unseal_and_spill_test.cpp:186:8: error: ‘init.{anonymous}::item_initialization::fill_parent’ may be used uninitialized [-Werror=maybe-uninitialized]
  186 | struct item_initialization {
      |        ^~~~~~~~~~~~~~~~~~~
../tests/unseal_and_spill_test.cpp: In function ‘std::pair<{anonymous}::item_initialization, std::__cxx11::basic_string<char> > {anonymous}::scenario_initial_state(scenario)’:
../tests/unseal_and_spill_test.cpp:310:25: note: ‘init’ declared here
  310 |     item_initialization init;
      |                         ^~~~
In copy constructor ‘{anonymous}::item_initialization::item_initialization(const {anonymous}::item_initialization&)’,
    inlined from ‘constexpr std::pair<_T1, _T2>::pair(_U1&&, _U2&&) [with _U1 = {anonymous}::item_initialization&; _U2 = std::__cxx11::basic_string<char>&; typename std::enable_if<(std::_PCC<true, _T1, _T2>::_MoveConstructiblePair<_U1, _U2>() && std::_PCC<true, _T1, _T2>::_ImplicitlyMoveConvertiblePair<_U1, _U2>()), bool>::type <anonymous> = true; _T1 = {anonymous}::item_initialization; _T2 = std::__cxx11::basic_string<char>]’ at /usr/include/c++/14/bits/stl_pair.h:882:4,
    inlined from ‘std::pair<{anonymous}::item_initialization, std::__cxx11::basic_string<char> > {anonymous}::scenario_initial_state(scenario)’ at ../tests/unseal_and_spill_test.cpp:365:29:
Error: ../tests/unseal_and_spill_test.cpp:186:8: error: ‘init.{anonymous}::item_initialization::seal’ may be used uninitialized [-Werror=maybe-uninitialized]
  186 | struct item_initialization {
      |        ^~~~~~~~~~~~~~~~~~~
../tests/unseal_and_spill_test.cpp: In function ‘std::pair<{anonymous}::item_initialization, std::__cxx11::basic_string<char> > {anonymous}::scenario_initial_state(scenario)’:
../tests/unseal_and_spill_test.cpp:310:25: note: ‘init’ declared here
  310 |     item_initialization init;
      |                         ^~~~
```
examples of failures on two unrelated pull requests, the second one is just some json changes
https://github.com/CleverRaven/Cataclysm-DDA/actions/runs/22183929368/job/64180206856?pr=85181
https://github.com/CleverRaven/Cataclysm-DDA/actions/runs/22194361217/job/64191034118?pr=85413

### Attach save file

N/A

### Steps to reproduce

N/A

### Expected behavior

N/A

### Screenshots

_No response_

### Versions and configuration

N/A

### Additional context

_No response_

--- a/tests/unseal_and_spill_test.cpp
+++ b/tests/unseal_and_spill_test.cpp
@@ -185,8 +185,8 @@ void unseal_items_containing( contents_change_handler &handler, item_location &r
 
 struct item_initialization {
     itype_id id;
-    bool fill_parent;
-    bool seal;
+    bool fill_parent = false;
+    bool seal = false;
     std::vector<item_initialization> contents;
 };
 

